std/unicode.zig

```diff
@@ -17,12 +17,29 @@ pub fn utf8CodepointSequenceLength(c: u32) !u3 {
 /// Given the first byte of a UTF-8 codepoint,
 /// returns a number 1-4 indicating the total length of the codepoint in bytes.
 /// If this byte does not match the form of a UTF-8 start byte, returns Utf8InvalidStartByte.
-pub fn utf8ByteSequenceLength(first_byte: u8) !u3 {
-    if (first_byte < 0b10000000) return u3(1);
-    if (first_byte & 0b11100000 == 0b11000000) return u3(2);
-    if (first_byte & 0b11110000 == 0b11100000) return u3(3);
-    if (first_byte & 0b11111000 == 0b11110000) return u3(4);
-    return error.Utf8InvalidStartByte;
+const Utf8ByteSequenceLengthError = error{
+    Utf8InvalidStartByte,
+};
+pub fn utf8ByteSequenceLength(first_byte: u8) Utf8ByteSequenceLengthError!u3 {
+    var ret: u4 = undefined;
+    ret = utf8ByteSequenceLengthNoValidate(first_byte);
+    if (ret > 4)
+        return error.Utf8InvalidStartByte;
+    return @intCast(u3, ret);
+}
+
+pub fn utf8ByteSequenceLengthNoValidate(first_byte: u8) u4 {
+    const utf8_skip_data = []u8{
+      1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,
+      1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,
+      1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,
+      1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,
+      1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,
+      1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,
+      2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,
+      3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,3,4,4,4,4,4,4,4,4,5,5,5,5,6,6,7,8, // Using \xfe and \xff allows utf-8 to encode 42-bits
```

Comment:
As i mentioned above, this is not the correct definition of UTF-8.

Reply:
It is fine, as long as you test for len > 4 (utf-8 can encode up to 42-bits, but that is not Unicode, they are two differn't standards) But I wrote a much more obvious version with `@clz()` that I think you will like.`

Reply:
The skip data comes from a totally differn't code set, because while the validate code doesn't need it, I need this information.